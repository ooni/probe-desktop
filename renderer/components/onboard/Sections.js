import React, { useCallback, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { MdKeyboardArrowLeft } from 'react-icons/md'
/*
 * XXX This component is a bit written a bit hasily. State is stored and encapsulated in the following way:
 *
 * The `Sections` component keeps track of:
 * - What is the current step of the wizard
 * - If the Quiz has been completed and/or if it's shown or hidden

 * The `QuizSteps` component is responsible for the state of:
 * - What is the current step of the quizActive
 *
 * It learns of if it should be shown or hidden from the props passed to it by
 * the Sections component.
 *
 * Using something like flux would have maybe made writing this easier, but this
 * is really the only place in which I felt the need of something like flux.
 */

import {
  Button,
  Box,
  Flex,
  Heading
} from 'ooni-components'
import NLink from 'next/link'

import FormattedMarkdownMessage from '../FormattedMarkdownMessage'
import ExternalLink from '../ExternalLink'
import Stepper from './Stepper'
import QuizSteps from './QuizSteps'

const HeadsUpList = styled.li`
  margin-top: 20px;
  margin-bottom: 20px;
`

const TopBar = styled(Box)`
  position: absolute;
  top: 0;
  width: 100%;
  height: 50px;
  background-color: transparent;
  color: ${props => props.theme.colors.white};
  /* This makes it possible to drag the window around from the side bar */
  -webkit-app-region: drag;
`

const OnboardBG = styled(Flex)`
  background: ${props => `no-repeat url(${props.img})`};
  background-color: ${props => props.bgColor || '#002b54'};
  background-size: contain;
  height: 100vh;
`

const SectionThingsToKnow = ({onNext, quizActive, quizComplete, toggleQuiz, onQuizComplete}) => (
  <div>
    {(quizActive && !quizComplete) &&
      <QuizSteps
        onClose={toggleQuiz}
        onDone={onQuizComplete}
        questionList={[
          <FormattedMessage key='Onboarding.PopQuiz.1.Question' id='Onboarding.PopQuiz.1.Question' />,
          <FormattedMessage key='Onboarding.PopQuiz.2.Question' id='Onboarding.PopQuiz.2.Question' />
        ]}
        actuallyList={[
          <FormattedMessage key='Onboarding.PopQuiz.1.Wrong.Paragraph' id='Onboarding.PopQuiz.1.Wrong.Paragraph' />,
          <FormattedMessage key='Onboarding.PopQuiz.2.Wrong.Paragraph' id='Onboarding.PopQuiz.2.Wrong.Paragraph' />
        ]}
      />
    }

    <Flex flexWrap='wrap' flexDirection='column'>
      <Box width={1}>
        <Heading textAlign='center' h={1}>
          <FormattedMessage id="Onboarding.ThingsToKnow.Title" />
        </Heading>
      </Box>
      <Box width={2/3} my={0} mx='auto'>
        <ul>
          <HeadsUpList><FormattedMessage id="Onboarding.ThingsToKnow.Bullet.1" /></HeadsUpList>
          <HeadsUpList><FormattedMessage id="Onboarding.ThingsToKnow.Bullet.2" /></HeadsUpList>
          <HeadsUpList><FormattedMessage id="Onboarding.ThingsToKnow.Bullet.3" /></HeadsUpList>
        </ul>
      </Box>
      <Box mx='auto'>
        <Button inverted onClick={quizComplete ? onNext : toggleQuiz}>
          <FormattedMessage id="Onboarding.ThingsToKnow.Button" />
        </Button>
      </Box>
      <Box mt={3} mx='auto'>
        <NLink href='https://ooni.org/about/risks/' passHref>
          <ExternalLink color='gray3'><FormattedMessage id="Settings.About.Content.LearnMore" /></ExternalLink>
        </NLink>
      </Box>
    </Flex>
  </div>
)

SectionThingsToKnow.propTypes = {
  quizActive: PropTypes.bool,
  quizComplete: PropTypes.bool,
  toggleQuiz: PropTypes.func,
  onQuizComplete: PropTypes.func,
  onNext: PropTypes.func
}

const SectionWhatIsOONI = ({onNext}) => (
  <Flex flexWrap='wrap' flexDirection='column'>
    <Box width={1} px={4}>
      <Heading h={1} textAlign='center'>
        <FormattedMessage id="Onboarding.WhatIsOONIProbe.Title" />
      </Heading>
    </Box>
    <Box width={1/2} my={3} px={4} mx='auto'>
      <FormattedMarkdownMessage id="Onboarding.WhatIsOONIProbe.Paragraph" />
    </Box>
    <Box mx='auto'>
      <Button inverted onClick={onNext} data-testid='got-it-button'>
        <FormattedMessage id='Onboarding.WhatIsOONIProbe.GotIt' />
      </Button>
    </Box>
  </Flex>
)

SectionWhatIsOONI.propTypes = {
  onNext: PropTypes.func
}

const NoButton = styled(Button)`
  color: white;
  border-color: white;
  &:hover:enabled {
    border-color: ${props => props.theme.colors.gray0};
  }
`

const SectionCrashReporting = ({ onOptIn, onOptOut }) => (
  <Flex flexDirection='column' alignItems='center'>
    <Box>
      <Heading h={1}>
        <FormattedMessage id='Onboarding.Crash.Title' />
      </Heading>
    </Box>
    <Box width={1/2} my={3} px={4}>
      <FormattedMarkdownMessage id='Onboarding.Crash.Paragraph' />
    </Box>
    <Flex flexDirection='row' alignItems='center' justifyContent='center' my={4}>
      <NoButton mx={2} hollow onClick={onOptOut}><FormattedMessage id='Onboarding.Crash.Button.No' /></NoButton>
      <Button mx={2} inverted onClick={onOptIn}><FormattedMessage id='Onboarding.Crash.Button.Yes' /></Button>
    </Flex>
  </Flex>
)
SectionCrashReporting.propTypes = {
  onOptIn: PropTypes.func,
  onOptOut: PropTypes.func
}

const SectionDefaultSettings = ({ onGo }) => (
  <Flex flexDirection='column'>
    <Box>
      <Heading textAlign='center' h={1}>
        <FormattedMessage id='Onboarding.DefaultSettings.Title' />
      </Heading>
    </Box>
    <Box px={4}>
      <Flex>
        <Box width={1/2}>
          <Heading h={4}>
            <FormattedMessage id='Onboarding.DefaultSettings.Header' />
          </Heading>
          <ul>
            <li>
              <FormattedMessage id='Onboarding.DefaultSettings.Bullet.1' />
            </li>
            <li>
              <FormattedMessage id='Onboarding.DefaultSettings.Bullet.2' />
            </li>
            <li>
              <FormattedMessage id='Onboarding.DefaultSettings.Bullet.3' />
            </li>
          </ul>
        </Box>
        <Box width={1/2} mt={4}>
          <FormattedMarkdownMessage id='Onboarding.DefaultSettings.Paragraph' />
        </Box>
      </Flex>
    </Box>
    <Flex alignItems='center' flexDirection='column'>
      <Button inverted onClick={onGo} data-testid='letsgo'>
        <FormattedMessage id='Onboarding.DefaultSettings.Button.Go' />
      </Button>
    </Flex>
  </Flex>
)

SectionDefaultSettings.propTypes = {
  onGo: PropTypes.func,
  onChange: PropTypes.func
}

const BackButtonContainer = styled.div`
  position: absolute;
  left: 10px;
  top: 40px;
  cursor: pointer;
  &:hover {
    color: ${props => props.theme.colors.gray4};
  }
`

const numSteps = 4

const onboardingBGs = [
  '/static/images/onboarding_0.svg',
  '/static/images/onboarding_1.svg',
  '/static/images/onboarding_2.svg',
  '/static/images/onboarding_2.svg'
]

const Sections = ({ onGo }) => {
  const [activeSection, setActiveSection] = useState(0)
  const [quizComplete, setQuizComplete] = useState(false)
  const [quizActive, setQuizActive] = useState(false)
  const [crashReportsOptIn, setOptIn] = useState(false)

  const toggleQuiz = useCallback(() => {
    setQuizActive(quizActive => !quizActive)
  }, [setQuizActive])

  const onQuizComplete = useCallback(() => {
    setQuizComplete(true)
    setActiveSection(activeIdx => activeIdx + 1)
  }, [setQuizComplete, setActiveSection])

  const nextStep = useCallback(() => {
    if (activeSection >= numSteps) {
      return
    }

    setActiveSection(activeSection + 1)
  }, [activeSection, setActiveSection ])

  const prevStep = useCallback(() => {
    if (activeSection <= 0) {
      return
    }

    setActiveSection(activeSection - 1)
  }, [activeSection, setActiveSection ])

  return (
    <OnboardBG
      img={onboardingBGs[activeSection]}
      // last onboarding screen needs a darker background
      bgColor={activeSection === 3 ? '#001a33' : '#002b54'}
      flexDirection='column'
      justifyContent='flex-end'
      flex='0 1'
    >
      <TopBar />
      <Flex flexWrap='wrap' justifyContent='center' alignItems='center'>
        <Box width={1}>
          {activeSection === 0 &&
            <SectionWhatIsOONI
              onNext={nextStep}
            />
          }

          {activeSection === 1 &&
            <SectionThingsToKnow
              onQuizComplete={onQuizComplete}
              quizComplete={quizComplete}
              quizActive={quizActive}
              toggleQuiz={toggleQuiz}
              onPrevious={prevStep}
              onNext={nextStep}
            />
          }
          {activeSection === 2 &&
            <SectionCrashReporting
              onOptIn={() => {
                setOptIn(true)
                nextStep()
              }}
              onOptOut={() => {
                setOptIn(false)
                nextStep()
              }}
            />
          }

          {activeSection === 3 &&
            <SectionDefaultSettings
              onGo={() => onGo(crashReportsOptIn)}
            />
          }
        </Box>

        <Box width={1}>
          {activeSection !== 0 &&
            <BackButtonContainer onClick={prevStep}>
              <Flex>
                <Box><MdKeyboardArrowLeft size={20} /></Box>
                <Box><FormattedMessage id='Onboarding.PopQuiz.Wrong.Button.Back' /></Box>
              </Flex>
            </BackButtonContainer>
          }
        </Box>

        <Box width={1} my={4}>
          <Stepper activeIdx={activeSection} />
        </Box>
      </Flex>
    </OnboardBG>
  )
}

Sections.propTypes = {
  onGo: PropTypes.func,
}

export default Sections
