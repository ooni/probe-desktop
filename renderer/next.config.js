// const withSourceMaps = require('@zeit/next-source-maps')
// const withBundleAnalyzer = require('@next/bundle-analyzer')({
//   enabled: process.env.ANALYZE === 'true'
// })

// module.exports = {
//   // node: {global: true},
//   webpack: (config, options) => {
//     config.target = 'electron-renderer'

//     config.node = {
//       __dirname: false,
//       __filename: false,
//       global: true
//     }

//     // config.plugins.push(
//     //   new options.webpack.DefinePlugin({
//     //     'process.env.NEXT_IS_SERVER': JSON.stringify(
//     //       options.isServer.toString()
//     //     )
//     //   })
//     // )

//     // config.module.rules.push({
//     //   test: /\.(eot|ttf|woff|woff2|otf)$/,
//     //   use: [
//     //     {
//     //       loader: 'url-loader'
//     //     }
//     //   ]
//     // })
//     // if (!options.isServer) {
//     //   config.resolve.alias['@sentry/node'] = '@sentry/browser'
//     // }
//     return config
//   },
// }

const nextConfig = {
  output: 'export',
  webpack: (config) => {
    config.module.rules.push({
      test: /\.svg$/,
      use: ['@svgr/webpack'],
    });
    return config;
  },
}

module.exports = nextConfig
