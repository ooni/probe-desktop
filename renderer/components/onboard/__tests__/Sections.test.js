import React from 'react'
import { screen, render, fireEvent, waitForElementToBeRemoved } from '@testing-library/react'
import { theme } from 'ooni-components'
import { ThemeProvider } from 'styled-components'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import English from '../../../../lang/en.json'

import Sections from '../Sections'

const renderComponent = (component, locale = 'en', messages = English) => {
  return render(
    <IntlProvider locale={locale} messages={messages}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </IntlProvider>
  )
}

// const rgb2hex = (rgb) => `#${rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/).slice(1).map(n => parseInt(n, 10).toString(16).padStart(2, '0')).join('')}`

describe('Tests for Screen 1 of Sections component', () => {
  test('Matches snapshot', async () => {
    const component = renderer
      .create(
        <IntlProvider locale='en' messages={English}>
          <ThemeProvider theme={theme}>
            <Sections />
          </ThemeProvider>
        </IntlProvider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
  test('Brings up the next screen', async () => {
    renderComponent(<Sections />)
    const gotItButton = screen.getByRole('button', { name: English['Onboarding.WhatIsOONIProbe.GotIt'] })
    fireEvent.click(gotItButton)
    const page2Title = await screen.findByRole('heading', { name: English['Onboarding.ThingsToKnow.Title'] })
    expect(page2Title).toBeInTheDocument()
  })
  // test('CSS styles of elements are correct', async () => {
  //   renderComponent(<Sections />)
  //   const gotItButton = screen.getByRole('button', { name: English['Onboarding.WhatIsOONIProbe.GotIt'] })
  //   const gotItButtonStyles = global.getComputedStyle(gotItButton)
  //   expect(rgb2hex(gotItButtonStyles.backgroundColor)).toMatch(theme.colors.white)
  //   expect(rgb2hex(gotItButtonStyles.color)).toMatch(theme.colors.primary)
  // })
})

const setup = () => {
  renderComponent(<Sections />)
  const gotItButton = screen.getByRole('button', { name: English['Onboarding.WhatIsOONIProbe.GotIt'] })
  fireEvent.click(gotItButton)
}

describe('Tests for Screen 2 of Sections component', () => {
  test('Screen 2 renders content correctly', async () => {
    setup()
    const page2Title = screen.getByRole('heading', { name: English['Onboarding.ThingsToKnow.Title'] })
    const mainButton = screen.getByRole('button', { name: English['Onboarding.ThingsToKnow.Button'] })
    const goBackButton = screen.getByText(English['Onboarding.PopQuiz.Wrong.Button.Back'])
    const learnMoreButton = screen.getByText(English['Onboarding.ThingsToKnow.LearnMore'])
    expect(page2Title).toBeInTheDocument()
    expect(mainButton).toBeInTheDocument()
    expect(goBackButton).toBeInTheDocument()
    expect(learnMoreButton).toBeInTheDocument()
    expect(learnMoreButton.href).toBe('https://ooni.org/about/risks/')
  })
  test('The *Go back* link brings back screen 1', async () => {
    setup()
    const goBackButton = screen.getByText(English['Onboarding.PopQuiz.Wrong.Button.Back'])
    fireEvent.click(goBackButton)
    const page1Title = await screen.findByRole('heading', { name: English['Onboarding.WhatIsOONIProbe.Title'] })
    expect(page1Title).toBeInTheDocument()
  })
  test('Clicking on \'I Understand\' brings up Pop Quiz', async () => {
    setup()
    const mainButton = screen.getByRole('button', { name: English['Onboarding.ThingsToKnow.Button'] })
    fireEvent.click(mainButton)
    const quizTitle = await screen.findByRole('heading', { name: English['Onboarding.PopQuiz.Title'] })
    expect(quizTitle).toBeInTheDocument()
  })
  // test('CSS styles of elements are correct', async () => {
  //   setup()
  //   const mainButton = screen.getByRole('button', { name: English['Onboarding.ThingsToKnow.Button'] })
  //   expect(mainButton).toHaveStyle(`background-color: ${theme.colors.white}`)
  //   expect(mainButtonStyles.color).toMatch(theme.colors.primary)
  // })
})

describe('Tests for Screens 3 and Screen 4 of Sections component', () => {
  test('Screen 3 and Screen 4 render correctly', async () => {
    const onGo = jest.fn()
    renderComponent(<Sections onGo={ onGo } />)

    const gotItButton = screen.getByRole('button', { name: English['Onboarding.WhatIsOONIProbe.GotIt'] })
    fireEvent.click(gotItButton)

    const screen2Button = screen.getByRole('button', { name: English['Onboarding.ThingsToKnow.Button'] })
    fireEvent.click(screen2Button)

    const yesButton1 = screen.getByText(English['Onboarding.PopQuiz.True'])
    fireEvent.click(yesButton1)

    const lottiePlayerFirst = screen.getByTestId('quiz-steps-tick')
    await waitForElementToBeRemoved(lottiePlayerFirst, { timeout: 2000 })

    const yesButton2 = screen.getByText(English['Onboarding.PopQuiz.True'])
    fireEvent.click(yesButton2)
    
    const lottiePlayerSecond = screen.getByTestId('quiz-steps-tick')
    await waitForElementToBeRemoved(lottiePlayerSecond, { timeout: 2000 })
    
    const popQuizTitle = screen.queryByText(English['Onboarding.PopQuiz.Title'])
    expect(popQuizTitle).not.toBeInTheDocument()
    
    const crashInfoTitle = await screen.findByText(English['Onboarding.Crash.Title'])
    expect(crashInfoTitle).toBeInTheDocument()
    const yesButton = screen.getByText(English['Onboarding.Crash.Button.Yes'])
    fireEvent.click(yesButton)

    const defaultSettingsTitle = await screen.findByText(English['Onboarding.DefaultSettings.Title'])
    expect(defaultSettingsTitle).toBeInTheDocument()
    const letsGoButton = screen.getByText(English['Onboarding.DefaultSettings.Button.Go'])
    fireEvent.click(letsGoButton)

    expect(onGo).toHaveBeenCalledTimes(1)
  })
})