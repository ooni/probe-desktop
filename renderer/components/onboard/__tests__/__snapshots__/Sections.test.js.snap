// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Screen 1 of Sections component Matches snapshot 1`] = `
.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  padding-left: 32px;
  padding-right: 32px;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  line-height: 1.5;
  font-weight: 300;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  font-size: 48px;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: center;
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 50%;
  margin-top: 16px;
  margin-bottom: 16px;
  padding-left: 32px;
  padding-right: 32px;
  margin-left: auto;
  margin-right: auto;
}

.c8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
}

.c9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 32px;
  padding-right: 32px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #0588cb;
  background-color: #f8f9fa;
  border-radius: 20px;
  border-width: 2px;
  font-size: 16px;
}

.c9:hover:enabled {
  background-color: #f1f3f5;
}

.c9:enabled {
  cursor: pointer;
}

.c10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  margin-top: 32px;
  margin-bottom: 32px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  top: 0;
  width: 100%;
  height: 50px;
  background-color: transparent;
  color: #f8f9fa;
  -webkit-app-region: drag;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex: 0 1;
  -ms-flex: 0 1;
  flex: 0 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: no-repeat url(/static/images/onboarding_0.svg);
  background-color: #002b54;
  background-size: contain;
  height: 100vh;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
      width={1}
    >
      <div
        className="c4"
      >
        <div
          className="c5"
          width={1}
        >
          <h1
            className="c6"
          >
            What is OONI Probe?
          </h1>
        </div>
        <div
          className="c7"
          width={0.5}
        >
          <p>
            Your app for measuring internet censorship. 
          </p>
          

          <p>
            Are websites and social media apps blocked? Is your internet connection unusually slow?
          </p>
          

          <p>
            Run OONI Probe to find out!
          </p>
        </div>
        <div
          className="c8"
        >
          <button
            className="c9"
            data-testid="got-it-button"
            fontSize={1}
            onClick={[Function]}
          >
            Got It
          </button>
        </div>
      </div>
    </div>
    <div
      className="c3"
      width={1}
    />
    <div
      className="c10"
      width={1}
    >
      <div
        className="c11"
      >
        <div
          className="c12"
        />
        <div
          className="c13"
        />
        <div
          className="c14"
        />
        <div
          className="c13"
        />
        <div
          className="c14"
        />
        <div
          className="c13"
        />
        <div
          className="c14"
        />
      </div>
    </div>
  </div>
</div>
`;
